src/MongoDB.Driver.Core/Core/WireProtocol/Messages/Encoders/BinaryEncoders/BinaryMessageEncoderFactory.cs

```diff
@@ -103,5 +109,11 @@ public IMessageEncoder GetUpdateMessageEncoder()
         {
             return new UpdateMessageBinaryEncoder(_stream, _encoderSettings);
         }
+
+        /// <inheritdoc />
+        public IMessageEncoder GetCompressedMessageEncoder()
+        {
+            return new CompressedMessageBinaryEncoder(_stream, _compressorSource, _encoderSettings);
+        }
```

Comment:
I think we need to pass the originalEncoderSelector through:

        public IMessageEncoder GetCompressedMessageEncoder(IMessageEncoderSelector originalEncoderSelector)
        {
            return new CompressedMessageBinaryEncoder(_stream, originalEncoderSelector, _compressorSource, _encoderSettings);
        }

Reply:
done